### PR TITLE
[SPARK-15189][PySpark][DOCS] Update ml.evaluation PyDoc 

### PR DESCRIPTION
**Pull request by holdenk**
## What changes were proposed in this pull request?

Fix doctest issue, short param description, and tag items as Experimental
## How was this patch tested?

build docs locally & doctests

### CODE REVIEW

**Comment by MLnick** [comment]
@holdenk what was the "doc test issue"? Is it the missing space?

**Comment by holdenk** [comment]
@MLnick indeed - the missing space kept the doctest from running & also screwed up the formatting in sphinx.

File: python/pyspark/ml/evaluation.py [hunk 1]
```diff
@@ -193,7 +197,11 @@ class RegressionEvaluator(JavaEvaluator, HasLabelCol, HasPredictionCol):
     # when we evaluate a metric that is needed to minimize (e.g., `"rmse"`, `"mse"`, `"mae"`),
     # we take and output the negative of this metric.
     metricName = Param(Params._dummy(), "metricName",
-                       "metric name in evaluation (mse|rmse|r2|mae)",
+                       """metric name in evaluation - one of:
+                       rmse - default root mean squared error
+                       mse - mean squared error
+                       r2 - r^2 metric
+                       mae - mean absolute error""",
```
Comment (MLnick) [inline python/pyspark/ml/evaluation.py]: minor, but perhaps add a `.` at the end

**Comment by MLnick** [comment]
Minor comment otherwise LGTM

File: python/pyspark/ml/evaluation.py [hunk 2]
```diff
@@ -193,7 +197,11 @@ class RegressionEvaluator(JavaEvaluator, HasLabelCol, HasPredictionCol):
     # when we evaluate a metric that is needed to minimize (e.g., `"rmse"`, `"mse"`, `"mae"`),
     # we take and output the negative of this metric.
     metricName = Param(Params._dummy(), "metricName",
-                       "metric name in evaluation (mse|rmse|r2|mae)",
+                       """metric name in evaluation - one of:
+                       rmse - default root mean squared error
```
Comment (MLnick) [inline python/pyspark/ml/evaluation.py]: @holdenk sorry I missed this before, but could we do it like `rmse - root mean squared error (default)` instead?

Comment (sethah) [inline python/pyspark/ml/evaluation.py]: Could you changes this to `rmse (default) - root mean squared error`? Seems a bit confusing the way it's worded now.

**Comment by MLnick** [comment]
Thanks! Merged to master and branch-2.0

**Pull request closed without merging**